src/components/Product/Product.js

```diff
@@ -1,43 +1,87 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import firebase from '@firebase/app';
+
 function Product() {
   const [option, setOption] = useState('');
   const [data, setData] = useState({
     name: '',
     lastDate: null,
   });
-  const addProduct = (event) => {
-    event.preventDefault();
-    console.log('send data...' + data.name, data.lastDate, option);
+
+  const [error, setError] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [productData, setProductData] = useState([]);
+
+  const viewMessage = (message, error) => {
+    setErrorMessage(message);
+    setError(error);
+  };
+
+  useEffect(() => {
     firebase
       .firestore()
-      .collection(token())
+      .collection('products')
+      .onSnapshot((snapshot) => {
+        let products = [];
+        snapshot.forEach((doc) => products.push(doc.data().name));
+        setProductData(products);
+      });
+  }, []);
+
+  const validateDuplicate = (product) => {
+    let productDataResult = productData;
+    if (productDataResult.indexOf(product) > -1) {
+      return true;
+    }
+    return false;
```

Comment:
Nice!

For future occasions, there's no need to do the `let productDataResult`. You could had done it like this

```
if (productData.indexOf(product) > -1) { ... }
```

Reply:
Hi Alejo, very thanks, for your comments, is corrected :) <3